src/object.d

```diff
@@ -2771,11 +2771,28 @@ private AARange _aaToRange(T: V[K], K, V)(ref T aa) pure nothrow @nogc @safe
 }
 
 /***********************************
- * Returns a forward range over the keys of the associative array.
+ * Returns a $(REF_ALTTEXT forward range, isForwardRange, std,range,primitives)
+ * which will iterate over the keys of the associative array. The keys are
+ * returned by reference.
+ *
+ * If changes are made to the associative array, all ranges previously obtained
+ * through this function are invalidated. The following example program will
+ * crash:
```

Comment:
Not exactly true. If you change just a value, the iteration will be fine.

Maybe say "If structural changes are made to the associative array (removing or adding keys), all ranges..."

This applies to all the other locations where you have this message.

I would also change "crash" to "dereference a null pointer" to clarify the behavior is not unsafe.